commands_to_load/Mod.py

```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        # Grab the last X messages from the channel regardless of user
+        logger.info('[Mod clear()] grabbing last {} message from {}'.format(numOfMsgs, channel))
+        messages = await channel.history(limit=numOfMsgs).flatten()
+        logger.info('[Mod clear()] messages to be deleted: {}'.format(messages))
+        
+        try:
+            await channel.delete_messages(messages)
+            logger.info('[Mod clear()] messages from {} deleted'.format(channel))
+        except discord.HTTPException:
+            eObj = await em(ctx, description='Messages cannot be older than 2 weeks', footer='Command Error')
+            if eObj is not False:
+                await ctx.send(embed=eObj, delete_after=10.0)
+            return
+
+        eObj = await em(ctx, description='{} messages deleted'.format(numOfMsgs), footer='Message will self destruct in 5 ...')
+        if eObj is not False:
+            await ctx.send(embed=eObj, delete_after=5.0)
+
+    @commands.command()
+    async def purge(self, ctx, *args):
+        # Deletes 10 message in channel from user
+        # Order of arguments doesn't matter, the code works around it
+        args = list(args)
+
+        # Verify Minion
+        logger.info('[Mod purge()] purge function detected by user' + str(ctx.message.author))
+        await ctx.message.delete()
+        logger.info('[Mod purge()] invoking command deleted')
+        
+        if not ctx.message.author in discord.utils.get(ctx.guild.roles, name="Minions").members:
+            logger.info('[Mod purge()] unathorized command attempt detected. Being handled.')
+            await self.rekt(ctx)
+            return
+
+        # Verify arguments
+        logger.info('[Mod purge()] verifying arguments')
+        mentions = ctx.message.mentions
+        if len(mentions) != 1: 
+            # There is no mentoin or more than 1
+            eObj = await em(ctx, description='Need to @ mention the user to purge messages from', footer='Invalid arguments')
+            if eObj is not False:
+                await ctx.send(embed=eObj)
+            return
+        elif len(mentions) == 1:
+            # Remove the mention from args and init the num var
+            user = mentions[0]
+            args.remove(user.mention)
+
+        if args:
+            num = int(args[0])
+        else:
+            num = 10
+        logger.info('[Mod purge()] all arguments verified')
+        logger.info('[Mod purge()] to purge messages from: {}\nNumber of messages to purge: {}'.format(user, num))
+
+        # Check command will determine if message is from user. Will need counter
+        def check(m):
+            nonlocal num
+            if m.author == user and num > 0:
+                num -=1
+                return True
+            else:
+                return False
+            
+        # Call channel.purge() limit at 100 and bulk = True
+        deleted = await ctx.channel.purge(limit=100, check=check, bulk=True)

```

Comment:
why place a limit of 100?

Reply:
originally i put that thinking nobody would deleted anywhere near that number but i've changed it so it while loops until the number is reached or theres no more messages to delete from the user.